src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -197,6 +212,20 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
     return this.applyFilters(filterItems, allItems);
   }
 
+  updateColumnFilter(checkboxValue: boolean, index: number) {
+    if (checkboxValue){
+      this.columnFilter = this.columnFilter | ( 1 << index );
+    } else {
+      this.columnFilter = this.columnFilter ^ ( 1 << index );
+    }
+  }
+
+  showColumn(index: number): boolean {
+    const {isConfigurable, tableId } = this.props;
+    if (!isConfigurable || !tableId) return true;
+    return ( this.columnFilter & 1 << index ) > 0;
+  }
```

Comment:
This current form is very susceptible to breaking (or at least odd behaviour) if we have `> 52` columns or if we ever rearrange the columns. The former is most likely a non-issue but the later is very much on the table. Since we plan on supporting re-sizable and rearrange-able columns in tables.

I am of the opinion that this feature should be designed with those in mind.